sdk/monitor/opentelemetry-exporter-azuremonitor/opentelemetry/exporter/azuremonitor/export/trace/__init__.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import json
+import logging
+from typing import Sequence
+from urllib.parse import urlparse
+
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+from opentelemetry.sdk.util import ns_to_iso_str
+from opentelemetry.trace import Span, SpanKind
+from opentelemetry.trace.status import StatusCanonicalCode
+
+from opentelemetry.exporter.azuremonitor import utils
+from opentelemetry.exporter.azuremonitor._generated.models import (
+    MonitorBase,
+    RemoteDependencyData,
+    RequestData,
+    TelemetryItem
+)
+from opentelemetry.exporter.azuremonitor.export import (
+    BaseExporter,
+    ExportResult,
+    get_trace_export_result,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AzureMonitorSpanExporter(BaseExporter, SpanExporter):
+    """Azure Monitor span exporter for OpenTelemetry.
+
+    Args:
+        options: :doc:`export.options` to allow configuration for the exporter
+    """
+
+    def __init__(self, **options):
+        super().__init__(**options)
+        self.add_telemetry_processor(indicate_processed_by_metric_extractors)
+
+    def export(self, spans: Sequence[Span]) -> SpanExportResult:
+        envelopes = list(map(self._span_to_envelope, spans))
```

Comment:
```suggestion
        envelopes = [self._span_to_envelope[span] for span in spans]
```
Let's avoid map and use list comprehensions instead. map is too javascripty.

Reply:
Suggested code trigger following error TypeError: 'method' object is not subscriptable

Reply:
updated - used the wrong brackets earlier ;)